libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java

```diff
@@ -1642,6 +1642,12 @@ public void test_multiple_aggregates_with_filter_clauses() {
             "FROM t");
     }
 
+    @Test
+    public void test_create_table_with_parametrized_varchar_data_type   () {
+        printStatement("create table test(col varchar(1))");
+        printStatement("create table test(col character varying(2))");
```

Comment:
If we keep the grammar as is (`dataType '('...`) I think we should also add tests for the array and object cases

Reply:
yes, it must be only for definedDataType and ident